activerecord/test/cases/associations/has_many_associations_test.rb

```diff
@@ -2578,22 +2578,22 @@ def test_association_with_rewhere_doesnt_set_inverse_instance_key
 
   test "association with extend option" do
     post = posts(:welcome)
-    assert_equal "lifo",  post.comments_with_extend.author
-    assert_equal "hello", post.comments_with_extend.greeting
+    assert_equal "lifo", post.comments_with_extend.author
+    assert_equal "hello :)", post.comments_with_extend.greeting
   end
 
   test "association with extend option with multiple extensions" do
     post = posts(:welcome)
-    assert_equal "lifo",  post.comments_with_extend_2.author
-    assert_equal "hullo", post.comments_with_extend_2.greeting
+    assert_equal "lifo", post.comments_with_extend_2.author
+    assert_equal "hullo :)", post.comments_with_extend_2.greeting
   end
 
   test "extend option affects per association" do
     post = posts(:welcome)
-    assert_equal "lifo",  post.comments_with_extend.author
-    assert_equal "lifo",  post.comments_with_extend_2.author
-    assert_equal "hello", post.comments_with_extend.greeting
-    assert_equal "hullo", post.comments_with_extend_2.greeting
+    assert_equal "lifo", post.comments_with_extend.author
+    assert_equal "lifo", post.comments_with_extend_2.author
+    assert_equal "hello :)", post.comments_with_extend.greeting
+    assert_equal "hullo :)", post.comments_with_extend_2.greeting
```

Comment:
https://github.com/rails/rails/pull/28830#discussion_r113633741

> The fact that you removed this and nothing breaks concerns me. This seems like something we need to do otherwise the association scope will not extend the module passed in the `:extend` option. Maybe we are missing test coverage. Can you check if this would make any difference in the public API?

I finally get your point. The extension block is used in `association.extensions` and `association.scope`.

https://github.com/rails/rails/blob/3bd534287139549cee10e10364344d66139d9237/activerecord/lib/active_record/associations/collection_proxy.rb#L29-L36

https://github.com/rails/rails/blob/3bd534287139549cee10e10364344d66139d9237/activerecord/lib/active_record/associations/association.rb#L139-L147

The former is simple and covered by the tests. Extracting the extension module just moves from `association.rb#L143` to `association.rb#L140` by this change. I've exercised the tests to ensure the extending order just in case.

The latter is hard to testing (the extension block can't affect to `eval_scope` loop both the former and the latter).
Extracting the extension block just moves from `association_scope.rb#136` to `association_scope.rb#28`.
 
https://github.com/rails/rails/blob/3bd534287139549cee10e10364344d66139d9237/activerecord/lib/active_record/associations/association_scope.rb#L24-L28

https://github.com/rails/rails/blob/3bd534287139549cee10e10364344d66139d9237/activerecord/lib/active_record/associations/association_scope.rb#L130-L137